Enable PT006 rule to airflow-core tests(triggers, timetables)

xchwan opened the pull request:
<!--
 Licensed to the Apache Software Foundation (ASF) under one
 or more contributor license agreements.  See the NOTICE file
 distributed with this work for additional information
 regarding copyright ownership.  The ASF licenses this file
 to you under the Apache License, Version 2.0 (the
 "License"); you may not use this file except in compliance
 with the License.  You may obtain a copy of the License at

   http://www.apache.org/licenses/LICENSE-2.0

 Unless required by applicable law or agreed to in writing,
 software distributed under the License is distributed on an
 "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 KIND, either express or implied.  See the License for the
 specific language governing permissions and limitations
 under the License.
 -->

<!--
Thank you for contributing! Please make sure that your code changes
are covered with tests. And in case of new features or big changes
remember to adjust the documentation.

Feel free to ping committers for the review!

In case of an existing issue, reference it using one of the following:

closes: #ISSUE
related: #ISSUE

How to write a good git commit message:
http://chris.beams.io/posts/git-commit/
-->



<!-- Please keep an empty line above the dashes. -->
---
Issue: Enable Even More PyDocStyle Checks #40567
@ferruzzi

This PR is for enable PT006 rule:
PT006: all instances of @pytest.mark.parametrize names should be a tuple
https://docs.astral.sh/ruff/rules/pytest-parametrize-names-wrong-type/
There are a lot of file changes needed.
So, I separate to many PR, which contain about 5-10 file changes for easy review.
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [airflow-core/newsfragments](https://github.com/apache/airflow/tree/main/airflow-core/newsfragments).

dependabot[bot] committed:
Bump the fab-ui-package-updates group across 1 directory with 2 updates (#57760)

* Bump the fab-ui-package-updates group across 1 directory with 2 updates

Bumps the fab-ui-package-updates group with 2 updates in the /providers/fab/src/airflow/providers/fab/www directory: [eslint](https://github.com/eslint/eslint) and [webpack-manifest-plugin](https://github.com/shellscape/webpack-manifest-plugin).


Updates `eslint` from 9.38.0 to 9.39.0
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.38.0...v9.39.0)

Updates `webpack-manifest-plugin` from 5.0.1 to 6.0.1
- [Release notes](https://github.com/shellscape/webpack-manifest-plugin/releases)
- [Commits](https://github.com/shellscape/webpack-manifest-plugin/compare/v5.0.1...v6.0.1)

---
updated-dependencies:
- dependency-name: eslint
  dependency-version: 9.39.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: fab-ui-package-updates
- dependency-name: webpack-manifest-plugin
  dependency-version: 6.0.1
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: fab-ui-package-updates
...

Signed-off-by: dependabot[bot] <support@github.com>

* Update hashes

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: vincbeck <vincbeck@amazon.com>

KoviAnusha committed:
Enable PT006 rule to 11 files in helm-tests (chart-utils,airflow_aux) (#57839)

Signed-off-by: Anusha Kovi <parvathi.kovi@gmail.com>
Co-authored-by: Kalyan R <kalyan.ben10@live.com>

xchwan committed:
enable PT011 (#57723)

Signed-off-by: Xch1 <qchwan@gmail.com>

xchwan committed:
modify test_deadline

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Xch1 <qchwan@gmail.com>

xchwan committed:
modify test_events_timetable

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Xch1 <qchwan@gmail.com>

KoviAnusha committed:
Enable PT006 rule to 12 files in helm-tests (airflow-core,apiserver,webserver) (#57838)

* Enable PT006 rule to 12 files in helm-tests (airflow-core,webserver,chart_utls)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Anusha Kovi <parvathi.kovi@gmail.com>

xchwan committed:
modify test_interval_timetable

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Xch1 <qchwan@gmail.com>

xchwan committed:
modify test_once_timetable

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Xch1 <qchwan@gmail.com>

xchwan committed:
modify test_trigger_timetable

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Xch1 <qchwan@gmail.com>

xchwan committed:
modify test_workday_timetable

Co-authored-by: Kalyan R <kalyan.ben10@live.com>
Signed-off-by: Xch1 <qchwan@gmail.com>

KoviAnusha committed:
Enable PT006 rule to dev (#57834)

Signed-off-by: Anusha Kovi <parvathi.kovi@gmail.com>
Co-authored-by: Kalyan R <kalyan.ben10@live.com>

ferruzzi reviewed (approved):
Approved once CI goes green

KoviAnusha committed:
Enable PT006 rule to kubernetes-tests, performance, airflow-ctl, docker-tests & shared (#57833)

Signed-off-by: Anusha Kovi <parvathi.kovi@gmail.com>
Co-authored-by: Kalyan R <kalyan.ben10@live.com>

ashb committed:
Ensure `task` in the context is always correct. (#57884)

Ensure `task` in the context is always correct.

ephraimbuddy committed:
Chart: Update default Airflow version to 3.1.2 (#57889)

amoghrajesh committed:
Ensure that DB migrations handles all kinds of NaN values in historical xcoms (#57866)

Ensure that DB migrations handles all kinds of NaN values in historical xcoms

dependabot[bot] committed:
Bump the core-ui-package-updates group across 1 directory with 16 updates (#57879)

Bumps the core-ui-package-updates group with 14 updates in the /airflow-core/src/airflow/api_fastapi/auth/managers/simple/ui directory:

| Package | From | To |
| --- | --- | --- |
| [@chakra-ui/react](https://github.com/chakra-ui/chakra-ui/tree/HEAD/packages/react) | `3.28.0` | `3.28.1` |
| [@tanstack/react-query](https://github.com/TanStack/query/tree/HEAD/packages/react-query) | `5.90.5` | `5.90.7` |
| [axios](https://github.com/axios/axios) | `1.13.0` | `1.13.2` |
| [react-hook-form](https://github.com/react-hook-form/react-hook-form) | `7.65.0` | `7.66.0` |
| [react-router-dom](https://github.com/remix-run/react-router/tree/HEAD/packages/react-router-dom) | `7.9.4` | `7.9.5` |
| [@eslint/compat](https://github.com/eslint/rewrite/tree/HEAD/packages/compat) | `1.4.0` | `1.4.1` |
| [@eslint/js](https://github.com/eslint/eslint/tree/HEAD/packages/js) | `9.38.0` | `9.39.1` |
| [@trivago/prettier-plugin-sort-imports](https://github.com/trivago/prettier-plugin-sort-imports) | `5.2.2` | `6.0.0` |
| [@vitejs/plugin-react-swc](https://github.com/vitejs/vite-plugin-react/tree/HEAD/packages/plugin-react-swc) | `4.2.0` | `4.2.1` |
| [@vitest/coverage-v8](https://github.com/vitest-dev/vitest/tree/HEAD/packages/coverage-v8) | `4.0.4` | `4.0.7` |
| [eslint](https://github.com/eslint/eslint) | `9.38.0` | `9.39.1` |
| [happy-dom](https://github.com/capricorn86/happy-dom) | `20.0.8` | `20.0.10` |
| [typescript-eslint](https://github.com/typescript-eslint/typescript-eslint/tree/HEAD/packages/typescript-eslint) | `8.46.2` | `8.46.3` |
| [vite](https://github.com/vitejs/vite/tree/HEAD/packages/vite) | `7.1.12` | `7.2.0` |



Updates `@chakra-ui/react` from 3.28.0 to 3.28.1
- [Release notes](https://github.com/chakra-ui/chakra-ui/releases)
- [Changelog](https://github.com/chakra-ui/chakra-ui/blob/main/packages/react/CHANGELOG.md)
- [Commits](https://github.com/chakra-ui/chakra-ui/commits/@chakra-ui/react@3.28.1/packages/react)

Updates `@tanstack/react-query` from 5.90.5 to 5.90.7
- [Release notes](https://github.com/TanStack/query/releases)
- [Changelog](https://github.com/TanStack/query/blob/main/packages/react-query/CHANGELOG.md)
- [Commits](https://github.com/TanStack/query/commits/@tanstack/react-query@5.90.7/packages/react-query)

Updates `axios` from 1.13.0 to 1.13.2
- [Release notes](https://github.com/axios/axios/releases)
- [Changelog](https://github.com/axios/axios/blob/v1.x/CHANGELOG.md)
- [Commits](https://github.com/axios/axios/compare/v1.13.0...v1.13.2)

Updates `react-hook-form` from 7.65.0 to 7.66.0
- [Release notes](https://github.com/react-hook-form/react-hook-form/releases)
- [Changelog](https://github.com/react-hook-form/react-hook-form/blob/master/CHANGELOG.md)
- [Commits](https://github.com/react-hook-form/react-hook-form/compare/v7.65.0...v7.66.0)

Updates `react-router-dom` from 7.9.4 to 7.9.5
- [Release notes](https://github.com/remix-run/react-router/releases)
- [Changelog](https://github.com/remix-run/react-router/blob/main/packages/react-router-dom/CHANGELOG.md)
- [Commits](https://github.com/remix-run/react-router/commits/react-router-dom@7.9.5/packages/react-router-dom)

Updates `@eslint/compat` from 1.4.0 to 1.4.1
- [Release notes](https://github.com/eslint/rewrite/releases)
- [Changelog](https://github.com/eslint/rewrite/blob/main/packages/compat/CHANGELOG.md)
- [Commits](https://github.com/eslint/rewrite/commits/compat-v1.4.1/packages/compat)

Updates `@eslint/js` from 9.38.0 to 9.39.1
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/commits/v9.39.1/packages/js)

Updates `@trivago/prettier-plugin-sort-imports` from 5.2.2 to 6.0.0
- [Release notes](https://github.com/trivago/prettier-plugin-sort-imports/releases)
- [Changelog](https://github.com/trivago/prettier-plugin-sort-imports/blob/main/CHANGELOG.md)
- [Commits](https://github.com/trivago/prettier-plugin-sort-imports/compare/v5.2.2...v6.0.0)

Updates `@vitejs/plugin-react-swc` from 4.2.0 to 4.2.1
- [Release notes](https://github.com/vitejs/vite-plugin-react/releases)
- [Changelog](https://github.com/vitejs/vite-plugin-react/blob/main/packages/plugin-react-swc/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite-plugin-react/commits/v4.2.1/packages/plugin-react-swc)

Updates `@vitest/coverage-v8` from 4.0.4 to 4.0.7
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v4.0.7/packages/coverage-v8)

Updates `eslint` from 9.38.0 to 9.39.1
- [Release notes](https://github.com/eslint/eslint/releases)
- [Commits](https://github.com/eslint/eslint/compare/v9.38.0...v9.39.1)

Updates `happy-dom` from 20.0.8 to 20.0.10
- [Release notes](https://github.com/capricorn86/happy-dom/releases)
- [Commits](https://github.com/capricorn86/happy-dom/compare/v20.0.8...v20.0.10)

Updates `typescript` from 5.9.3 to 5.8.3
- [Release notes](https://github.com/microsoft/TypeScript/releases)
- [Changelog](https://github.com/microsoft/TypeScript/blob/main/azure-pipelines.release-publish.yml)
- [Commits](https://github.com/microsoft/TypeScript/compare/v5.9.3...v5.8.3)

Updates `typescript-eslint` from 8.46.2 to 8.46.3
- [Release notes](https://github.com/typescript-eslint/typescript-eslint/releases)
- [Changelog](https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/typescript-eslint/CHANGELOG.md)
- [Commits](https://github.com/typescript-eslint/typescript-eslint/commits/v8.46.3/packages/typescript-eslint)

Updates `vite` from 7.1.12 to 7.2.0
- [Release notes](https://github.com/vitejs/vite/releases)
- [Changelog](https://github.com/vitejs/vite/blob/main/packages/vite/CHANGELOG.md)
- [Commits](https://github.com/vitejs/vite/commits/v7.2.0/packages/vite)

Updates `vitest` from 4.0.4 to 4.0.7
- [Release notes](https://github.com/vitest-dev/vitest/releases)
- [Commits](https://github.com/vitest-dev/vitest/commits/v4.0.7/packages/vitest)

---
updated-dependencies:
- dependency-name: "@chakra-ui/react"
  dependency-version: 3.28.1
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: "@tanstack/react-query"
  dependency-version: 5.90.7
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: axios
  dependency-version: 1.13.2
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: react-hook-form
  dependency-version: 7.66.0
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: react-router-dom
  dependency-version: 7.9.5
  dependency-type: direct:production
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: "@eslint/compat"
  dependency-version: 1.4.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: "@eslint/js"
  dependency-version: 9.39.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: "@trivago/prettier-plugin-sort-imports"
  dependency-version: 6.0.0
  dependency-type: direct:development
  update-type: version-update:semver-major
  dependency-group: core-ui-package-updates
- dependency-name: "@vitejs/plugin-react-swc"
  dependency-version: 4.2.1
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: "@vitest/coverage-v8"
  dependency-version: 4.0.7
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: eslint
  dependency-version: 9.39.1
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: happy-dom
  dependency-version: 20.0.10
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: typescript
  dependency-version: 5.8.3
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: typescript-eslint
  dependency-version: 8.46.3
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
- dependency-name: vite
  dependency-version: 7.2.0
  dependency-type: direct:development
  update-type: version-update:semver-minor
  dependency-group: core-ui-package-updates
- dependency-name: vitest
  dependency-version: 4.0.7
  dependency-type: direct:development
  update-type: version-update:semver-patch
  dependency-group: core-ui-package-updates
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

tuliolima committed:
Update logging information for callback errors (#57871)

bugraoz93 committed:
Release Notes change (#57895)

ramitkataria committed:
Refactor Deadline callbacks to use the new Callback model (#57215)

Followup to #54796. This PR essentially moves callback functionality
from deadline callbacks to the generic callbacks and refactors Deadline
to use these callbacks.

potiuk committed:
Fix pin-versions prek hook and upgrade hooks (#57857)

We had pin-versions prek-hook implemented in a separate workflow
under the `dev` folder, but it has not been working since workspace
switch because prek workspace only works on sub-folders of where the
.pre-commit-config.yaml file is placed. It was in a separate file
because it needed python 3.11 to run, but it is possible to have
a specific python verison as separate language version in the hook
itself, so we can move it back to the main .pre-commit-config.yaml

This PR:

* moves the pin-version hook to main .pre-commit-config.yaml
* sets python 3.11 as version of python used in the hook independently
  from default python version
* fix github actions and docs to use the hook from the main
  .pre-commit-config.yaml

potiuk committed:
Improve preparation and verification of source tarballs (#57901)

After testing with Alpha 2 ATR (Apache Trusted Releases) of airflow-ctl
it turned out that our source tarballs were not perfectly according to
what ATR expected.

* The -source tarballs did not have internal directory with `-source`
  sufffix

* The verification of those tarballs with RAT 0.17.0 was done using
  the .rat-excludes from airflow sources - not from the package itself

* For provider releases, we verified way more than we should with
  licence checks (we should really only verify the sources, not
  the whl or .sdist files.

The commands to generate the tarballs also had some inconsistencies and
repetitions:

* Literals were sprinkled across the code base

* In some cases we would produce tarballs when generating distributions
  which was not exactly matching the general "prepare-airflow-tarball"
  behaviour regarding tags and versions.

* It was difficult to test tarball preparation locally before tags
  were created for the version to test

* Tarballs should never be produced with rc suffix. We always
  produce tarballs with the "final version" of the component
  because tarballs are not published in PyPI and can be promoted
  to the final version always.

This PR simplifies the scripts and makes it more consistent:

* tarballs are only created with `prepare-tarball` command
* Literals are removed and replaced wiuth TarballType enum (DRY)
* behaviour of tarball generation is more predictable now and
  consistent:
  * you specify tarball type
  * if you do not specify version, tag is HEAD, and version is
    automatically retrieved based of the version of tarball type
  * if you specify version - prefix (airflow-ctl/, airflow-task-sdk/,
    providers/) is added to the version to derive a tag and that
    version is used as tarball version (prefixed with type)
  * tarball-type-version-source is used as top-level folder
    where the sources are placed
* verification of licences is now consistent, simplified and
  produces better results - showing counts of checked files and
  exclusions used - not only files that have unknown or unapproved
  licences. You can actually see that RAT did it's job.

Crowiant committed:
Remove deprecated in google provider scheduled for October 2025 (#56935)

Co-authored-by: Anton Nitochkin <nitochkin@google.com>

Crowiant committed:
Replace dag.log.info with log.info in system tests in google provider (#57550)

Co-authored-by: Anton Nitochkin <nitochkin@google.com>

alamashir committed:
Add gcp_conn_id parameter to DataprocDiagnoseClusterOperator examples (#57904)

* Fix systemd service files issues

This commit addresses multiple bugs in systemd service files reported
in issue #53706:

1. Fix incorrect command in airflow-api.service
   - Changed 'airflow api' to 'airflow api-server' (correct CLI command)

2. Standardize environment configuration approach
   - Changed airflow-api.service and airflow-triggerer.service to use
     EnvironmentFile=/etc/sysconfig/airflow instead of inline Environment
   - This provides consistency with other services and allows admins
     flexibility to manage custom environment variables centrally

3. Fix service dependencies
   - Added proper After= and Wants= clauses for database and message
     queue services (postgresql, mysql, redis, rabbitmq) to
     airflow-api.service and airflow-triggerer.service
   - Services now wait for required dependencies before starting

4. Additional bug fixes discovered during review
   - airflow-flower.service: Removed duplicate ExecStart line
   - airflow-kerberos.service: Fixed command from 'kerberos' to
     'airflow kerberos'

5. Consistency improvements
   - Added Type=simple and RestartSec=5s to airflow-api.service and
     airflow-triggerer.service for consistency with other services

All systemd service files now follow the same configuration pattern,
making them easier to maintain and more reliable during system startup.

Fixes #53706

* Add gcp_conn_id parameter to DataprocDiagnoseClusterOperator examples

Added gcp_conn_id parameter to both standard and deferrable examples
in the DataprocDiagnoseClusterOperator documentation to clarify that
users can customize the GCP connection ID.

Fixes: #44420

---------

Co-authored-by: Ashir Alam <ashiralam@Ashirs-MacBook-Pro.local>

KoviAnusha committed:
Enable PT006 rule to 9 files in providers (snowflake,smtp/tests) (#57845)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

KoviAnusha committed:
Enable PT006 rule to 15 files in helm-tests (other,security) (#57844)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

ephraimbuddy committed:
Airflow 3.1.2 has been released (#57887)

xchwan committed:
Enable PT006 rule to airflow-core tests (utils) (#57885)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

vincbeck committed:
Use Iterable instead of list in API responses (#57878)

xchwan committed:
Enable PT006 rule to task-sdk tests (#57841)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

xchwan committed:
Enable PT006 rule to airflow-core tests (security, lineage, jobs, executors, datasets) (#57913)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

dabla committed:
Make set_xcom route in API server DRY by reusing logic of XComModel.set method (#55289)

Co-authored-by: Tzu-ping Chung <uranusjr@gmail.com>

KoviAnusha committed:
Enable PT006 rule to 17 files in providers (operatorsproviders/amazon/tests/unit/amazon/aws/operators/) (#57903)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

pierrejeambrun committed:
Add number of queries guard for ui config (#57868)

xchwan committed:
Enable PT006 rule to airflow-core tests (ti_deps, serialization) (#57911)

Co-authored-by: Kalyan R <kalyan.ben10@live.com>

xchwan committed:
solve conflict

Signed-off-by: Xch1 <qchwan@gmail.com>
